docs/source/nodes_and_pipelines/nodes.md

```diff
@@ -163,6 +163,9 @@ kedro run --tags=pipeline_tag
 
 This will run only the nodes found within the pipeline tagged with `pipeline_tag`.
 
+```{note}
+Valid node name and/or tag must contain only letters, digits, hyphens, underscores and/or fullstops.
```

Comment:
@datajoely That looks good. If periods conflict with namespaces, isn't it better to that information to error message as well? https://github.com/kedro-org/kedro/blob/main/kedro/pipeline/node.py#L126-L139

Reply:
@datajoely I think this PR is better than not mentioning the limitation altogether, so should we just merge it in?

@astrojuanlu @merelcht @noklam wdyt?

Reply:
If a node_name contains a '.', there's a high chance it might be treated as if within a 'namespace'. We are currently encountering numerous issues on Kedro-Viz targeting nested modular pipelines that involve multiple '.' notations. If a node_name without a namespace contains a '.', then I have a feeling this might cause problems on Kedro-Viz.

I am conducting some tests next week on modular pipelines and will revert with more informed opinion once I have investigated this matter more thoroughly.

Reply:
My view is we either need to reserve `.` exclusively for namespaces or document the functionality explicitly, I'd be very reticent to consciously avoid mentioning

Reply:
Hi team,

With the recent refactor of Kedro-viz, the . notation for namespaces is now the primary and only indicator that a node has a namespace. The modular pipeline visualization also takes this into account. Tags do not have namespaces, and with this update, namespaces associated with datasets are no longer considered. Only node namespaces matter.

I agree with @datajoely that we should either exclusively reserve . for namespaces or find a public way for Kedro-viz to easily retrieve this information from Kedro.

Reply:
This also raises the question whether we should provide any validation on the front-end side because I know for a fact users use the dot syntax to simply organise hierarchies without actually using the `namespace=` argument.

Reply:
@datajoely - I thought the same, but now I am slighly confused as I tried to visualise a pipeline like the below that doesn't use modular_pipelines at all.  

```
def create_pipeline(**kwargs) -> Pipeline:
    return pipeline([
        
            node(
                lambda x: x,
                inputs=["raw_data"],
                outputs="model_inputs",
                name="uk.data_processing.process_data",
                tags=["split"],
            ),
            node(
                lambda x: x,
                inputs=["model_inputs"],
                outputs="model",
                name="uk.data_science.train_model",
                tags=["train"],
            )
        ])    
```

and I got the visualisation as the below, so no modular pipeline functionality despite using '.' notation
<img width="970" alt="Screenshot 2024-05-17 at 12 46 41" src="https://github.com/kedro-org/kedro/assets/37628668/9e058e60-f0cd-40bc-b586-a24860c2da63">

Reply:
@astrojuanlu I think I understood. I'll post here if I have any questions. Thanks.